src/renderers/webgl/WebGLProgram.js

```diff
@@ -701,7 +698,7 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 	vertexShader = unrollLoops( vertexShader );
 	fragmentShader = unrollLoops( fragmentShader );
 
-	if ( parameters.isWebGL2 ) {
+	if ( parameters.isWebGL2 && parameters.isRawShaderMaterial !== true ) {
```

Comment:
Did you test this with anisotropy enabled? I believe this will make it fail for WebGL2 because it doesn't translate `texture2DGradEXT` to `textureGrad`.

Reply:
> Did you test this with anisotropy enabled?  I believe this will make it fail for WebGL2 because it doesn't translate `texture2DGradEXT` to `textureGrad`.

Hmm, I have tested it with anisotropy enable yeah... 🤔

Reply:
Weird; I need to pull your branch and try this myself. I'm clearly misunderstanding something...

Reply:
I just pulled your branch and tested it with `anistropy: 16` on my macbook and it's definitely still broken.
<img width="851" alt="image" src="https://user-images.githubusercontent.com/1649964/155231973-e1ca76da-5b38-4d8d-882d-47a4c60d5d01.png">

Reply:
That's a different broken. I didn't test these settings as that's not how it used to break before.

Testing...

Reply:
I see... The `RawShaderMaterial` is missing the `#extension GL_EXT_shader_texture_lod : enable` line.